frontends/ast/genrtlil.cc

```diff
@@ -759,7 +759,7 @@ struct AST_INTERNAL::ProcessGenerator
 						arg.realtime = true;
 					} else {
 						arg.type = VerilogFmtArg::INTEGER;
-						arg.sig = node->genRTLIL();
+						arg.sig = node->genWidthRTLIL(-1, false, &subst_rvalue_map.stdmap());
```

Comment:
This is all that seems to need change. It introduces an extra call to `detectSignWidthWorker` from within `genWidthRTLIL` which hopefully is benign.